src/olympia/amo/utils.py

```diff
@@ -477,6 +478,40 @@ def parse_html(tree):
     return serializer.render(stream)
 
 
+def pngcrush_image(src, **kw):
+    """
+    Optimizes a PNG image by running it through Pngcrush.
+    """
+    log.info('Optimizing image: %s' % src)
+    try:
+        # When -ow is used, the output file name (second argument after
+        # options) is used as a temporary filename (that must reside on the
+        # same filesystem as the original) to save the optimized file before
+        # overwriting the original. By default it's "pngout.png" but we want
+        # that to be unique in order to avoid clashes with multiple tasks
+        # processing different images in parallel.
+        tmp_path = '%s.crush.png' % os.path.splitext(src)[0]
+        # -brute is not recommended, and in general does not improve things a
+        # lot. -reduce is on by default for pngcrush above 1.8.0, but we're
+        # still on an older version.
+        cmd = [settings.PNGCRUSH_BIN, '-q', '-reduce', '-ow', src, tmp_path]
+        process = subprocess.Popen(
+            cmd, stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
```

Comment:
I wonder why you're piping stdin here if you're not sending anything to the process?

Reply:
That was useless, removed in https://github.com/mozilla/addons-server/pull/7661/commits/52bd5606280ecd1c18193846fff0a1e2eb398a29